source/AI.cpp

```diff
@@ -769,8 +769,13 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			// A carried ship must belong to the same government as its parent to dock with it.
 			bool hasParent = parent && !parent->IsDestroyed() && parent->GetGovernment() == gov;
 			bool inParentSystem = hasParent && parent->GetSystem() == it->GetSystem();
+			// NPCs may take 30 seconds or longer to find a new parent.  Player
+			// owned fighter shouldn't take more than a few seconds.
+			bool findNewParent = it->IsYours() ? !Random::Int(30) : !Random::Int(1800);
 			bool parentHasSpace = inParentSystem && parent->BaysFree(it->Attributes().Category());
-			if(!hasParent || (!inParentSystem && !it->JumpFuel()) || (!parentHasSpace && !Random::Int(1800)))
+			if(findNewParent && parentHasSpace)
+				parentHasSpace = parent->CanCarry(*it);
```

Comment:
This expensive call is only called roughly twice a second.  Otherwise, the faster BaysFree call is used.  BaysFree is used to determine if this call should occur at all.

This enables fighters to choose different parents to return.  This faster path finding means from a player perspective abandoned fighters will in parallel choose a different parent which has free bays.